beacon-chain/utils/slot_ticker_test.go

```diff
@@ -0,0 +1,102 @@
+package utils
+
+import (
+	"testing"
+	"time"
+)
+
+func TestSlotTicker(t *testing.T) {
+	ticker := SlotTicker{
+		c:    make(chan uint64),
+		done: make(chan struct{}),
+	}
+	defer ticker.Done()
+
+	var sinceDuration time.Duration
+	since := func(time.Time) time.Duration {
+		return sinceDuration
+	}
+
+	var untilDuration time.Duration
+	until := func(time.Time) time.Duration {
+		return untilDuration
+	}
+
+	var tick chan time.Time
+	after := func(time.Duration) <-chan time.Time {
+		return tick
+	}
+
+	genesisTime := time.Date(2018, 1, 1, 0, 0, 0, 0, time.UTC)
+	slotDuration := uint64(8)
+
+	// Test when the ticker starts immediately after genesis time.
+	sinceDuration = 1 * time.Second
+	untilDuration = 7 * time.Second
+	// Make this a buffered channel to prevent a deadlock since
```

Comment:
Do punctuations like periods matter for test comments? If yes, you missed a few here : )